lib/tasks/scrape.rake

```diff
@@ -1,146 +1,159 @@
 namespace :scrape do 
   desc "Scraping Pinterest"
+
+  # must fill category table before doing this
   task :get_posts => :environment do
 
     require 'open-uri'
     require 'nokogiri'
     # require 'date'
 
+    if (Category.all.count == 0) then
+      puts "please run rake scrape:get_categories first"
+      return
+    end
 
-
-    # url = "http://pinterest.com/search/pins/?q=#{params[:q]}"
-    url = "https://www.pinterest.com/categories/popular/"
-    scrape_site(url)
-    # scrape_images(url)
+    Category.all.each do |category|
+      url = "https://www.pinterest.com" + category.link
+      puts url
+      # this is to pick up scraping 
+      # half way through in case of (404) error
+      # if (category.id > 20) then
+        scrape_site(url, category.id)
+      # end
+    end
   end
 
-  def scrape_site(url)
-
+  def scrape_site(url, categoryID)
     document = open(url).read
     html_doc = Nokogiri::HTML(document)
-    #puts html_doc
 
     data_pin_link = "a.pinImageWrapper"
     pin_links = html_doc.css(data_pin_link)
     puts "number of pin links " + pin_links.count.to_s
```

Comment:
Are we trying to scraping the pins on infinite scroll? Or are we just scraping all the pins on first load?